src/buffer/fences.rs

```diff
@@ -129,3 +129,43 @@ impl<'a> Inserter<'a> {
         *existing_fences = new_fences;
     }
 }
+
+/// An encapsulated storage of fence inserters
+/// Allows reusing a single Vec for fences without allocating new Vec's on every draw/compute call.
+pub struct FenceInserters<'a>(RefMut<'a, Vec<Option<Inserter<'a>>>>);
```

Comment:
Is the `Option` necessary here? Since from what I can gather it'll always be `Some`.